test/e2e/namespace.go

```diff
@@ -74,6 +75,57 @@ func extinguish(f *Framework, totalNS int, maxAllowedAfterDel int, maxSeconds in
 		}))
 }
 
+func ensurePodsAreRemovedWhenNamespaceIsDeleted(f *Framework) {
+	var err error
+
+	By("Creating a test namespace")
+	namespace, err := f.CreateNamespace("nsdeletetest", nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Waiting for a default service account to be provisioned in namespace")
+	err = waitForDefaultServiceAccountInNamespace(f.Client, namespace.Name)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Creating a pod in the namespace")
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name: "test-pod",
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:  "nginx",
+					Image: "gcr.io/google_containers/pause:2.0",
+				},
+			},
+		},
+	}
+	pod, err = f.Client.Pods(namespace.Name).Create(pod)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Waiting for the pod to have running status")
+	expectNoError(waitForPodRunningInNamespace(f.Client, pod.Name, pod.Namespace))
```

Comment:
@quinton-hoole - The priority for this test was to ensure that it captured graceful deletion waiting periods since I viewed that as most complicated at this time.  Just took a quick look at the namespace controller, and we do not actually differentiate pods based on their status when calculating graceful deletion estimates:

https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/namespace/namespace_controller_utils.go#L303

So from a test perspective, it most likely would not make a difference.  I think we can improve the namespace controller to not include the grace period buffer when pods are in a terminal state.

Opened an issue for that: https://github.com/kubernetes/kubernetes/issues/21701